.github/workflows/scorecard.yml

```diff
@@ -0,0 +1,73 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
+
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '0 6 * * 1' # 6 AM UTC every Monday
+  push:
+    branches: [ "master" ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
Nit: we don't typically pin to a specific SHA. I'd suggest not doing so here, or updating all other GHA usages to use a SHA, to be consistent. Otherwise contributors may be confused as to which is preferable. Even below in this workflow file there are usages of version tags rather than SHAs.

Reply:
Actually while doing this work, I discovered that SHAs is the preferable way of pinning because the version method doesn't guarantee an immutable release. On the other hand, I can see how this is inconsistent with the rest of the workflows.

Reply:
Changed in https://github.com/vectordotdev/vector/pull/21942/commits/866855f2a4f3e0c34817bc9ee8d342ff20051e72. We can re-consider this later holistically.